### PR TITLE
Integrated + n1

### PR DESCRIPTION
**Pull request by erickma1**
In this milestone, I did the following;

- Used Capybara to write integration tests for each view in my project.
- 
- Solve N+1 problems when fetching all posts and their comments for a user.

Screenshot before changes

before

Screenshot after changes

after

### CODE REVIEW

File: spec/integration/userposts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      expect(page).to have_content(post.title)
+      expect(page).to have_content(post.text.truncate(50))
+      expect(page).to have_content(post.comments.first.body) if post.comments.any?
+      expect(page).to have_content("Comments #{post.comments.count}")
+      expect(page).to have_content("Likes #{post.likes.count}") # Truncate post body for visibility
+    end
+  end
+  scenario 'Check if the pagination section is visible' do
+    expect(page).to have_css('.pagination')
+  end
```
Comment (Ibizugbe) [inline spec/integration/userposts_spec.rb:29]: Nice job so far👍🏼. As a requirement, all tests must be passed. kindly ensure all tests are passing before proceeding to submit😉.

image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @erickma1 👋🏼👋🏼👋🏼, 

request change


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Linters are passing👍🏼.
- [x] capybara gem is present in the gemfile👍🏼.
- [x] The respective tests for the user index and show page have been written👍🏼.
- [x] The respective tests for the post index and show page have been written👍🏼.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by erickma1** [comment]
corrected mistakes in tests

**Review by Munsa1 — APPROVED** [review]
Hi @erickma1, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Changes :heavy_check_mark: 
Correct Git Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**